source/parser/comptime.civet

```diff
@@ -139,10 +139,31 @@ function serialize(value: ???): string
             when Array::
               `[${(val as ???[]).map(recurse).join ","}]`
             when Object::
-              "{" + (
-                for own key, value in val as {[key: string]: ???}
-                  `${JSON.stringify key}:${recurse value}`
-              ).join(",") + "}"
+              // The string representing the object's own enumerable, writable properties
+              objStr .= '{'
+              // The string representing other property descriptors, if applicable, not including the outermost braces
+              descStr .= ''
+              descriptors := Object.getOwnPropertyDescriptors(val) as Record<string | symbol, PropertyDescriptor>
+              for prop of (Object.getOwnPropertyNames(val) as (string|symbol)[]) ++ Object.getOwnPropertySymbols(val)
```

Comment:
```suggestion
              for each prop of (Object.getOwnPropertyNames(val) as (string|symbol)[]) ++ Object.getOwnPropertySymbols(val)
```

Wow, enumeration is hard. I suppose we could call `Object.getOwnPropertyDescriptor` inside the loop instead of the plural form outside the loop, but this way may be more time-efficient (if a bit less memory efficient).

Reply:
Good point, I think I will do `getOwnPropertyDescriptor`, if only to make this a little easier to read.